config/v1/types_node.go

```diff
@@ -46,6 +46,17 @@ type NodeSpec struct {
 	// the status and corresponding reaction of the cluster
 	// +optional
 	WorkerLatencyProfile WorkerLatencyProfileType `json:"workerLatencyProfile,omitempty"`
+
+	// MinimumKubeletVersion is the lowest version of a kubelet that can meaningfully join the cluster.
+	// Specifically, the apiserver will deny most authorization requests of kubelets that are older
+	// than the specified version, only allowing the kubelet to get and update its node object, and perform
+	// subjectaccessreviews.
+	// Its max length is 8, so maximum version allowed is either "9.999.99" or "99.99.99".
```

Comment:
How was the maximum length decided? Is this a kubelet validation that we are transferring?

Reply:
I decided semi arbitrarily based on what felt like a reasonable maximum. honestly, I don't think there's the notion of maximum length of kubernetes version, but practically I'd be surprised if this API lives long enough to hit this limit (10 years has gotten us to 1.31.0)